operator-common/src/main/java/io/strimzi/operator/common/AbstractOperator.java

```diff
@@ -488,31 +488,38 @@ public AtomicInteger getResourceCounter() {
      * @param reconciliation reconciliation to use to update the resource state metric
      * @param ready if reconcile was successful and the resource is ready
      */
-    private void updateResourceState(Reconciliation reconciliation, boolean ready) {
-        Tags metricTags = Tags.of(
+    private void updateResourceState(Reconciliation reconciliation, boolean ready, Throwable cause) {
+        String key = reconciliation.namespace() + ":" + reconciliation.kind() + "/" + reconciliation.name();
+
+        Tags metricTagsWithReason = Tags.of(
                 Tag.of("kind", reconciliation.kind()),
                 Tag.of("name", reconciliation.name()),
-                Tag.of("resource-namespace", reconciliation.namespace()));
+                Tag.of("resource-namespace", reconciliation.namespace()),
+                Tag.of("failure-reason", cause == null ? "none" : cause.getMessage()));
```

Comment:
why did you change the key of the map, using a string with the same field unless failure-reason. Why it should not be part of the key in that map? The tags are a way to identify a metric making a kind of compound key. Why failure-reason should be out?

Reply:
If the failure reason changes, the key changes, and thus we have two different metrics.